### PR TITLE
Z-Mimic Fixes Pack 2

### PR DESCRIPTION
**Pull request by Lohikar**
Reworks layering a bit so it actually works. I couldn't get large turfs to play nice, so they get a flag that changes (read: breaks) the layering so they don't look bad.

~~DNM: Untested. It compiles and the O7 ver works, that's all I know.~~ Seems to work.

### CODE REVIEW

File: code/controllers/subsystems/zcopy.dm [hunk 1]
```diff
@@ -77,10 +81,23 @@ SUBSYSTEM_DEF(zcopy)
 	..("Q:{T:[queued_turfs.len - (qt_idex - 1)]|O:[queued_overlays.len - (qo_idex - 1)]} T:{T:[openspace_turfs]|O:[openspace_overlays]}")
 
 /datum/controller/subsystem/zcopy/Initialize(timeofday)
+	calculate_zstack_limits()
 	// Flush the queue.
 	fire(FALSE, TRUE)
 	return ..()
 
+// If you add a new Zlevel or change Z-connections, call this.
+/datum/controller/subsystem/zcopy/proc/calculate_zstack_limits()
```
Comment (Lohikar) [inline code/controllers/subsystems/zcopy.dm:90]: Not familiar with away site loading et-al, so not sure where this should get called.

File: code/controllers/subsystems/zcopy.dm [hunk 2]
```diff
@@ -127,28 +142,35 @@ SUBSYSTEM_DEF(zcopy)
...
+				if (!T.below.bound_overlay)
+					T.below.bound_overlay = new(T)
+				var/atom/movable/openspace/turf_overlay/TO = T.below.bound_overlay
+				TO.appearance = T.below
+				TO.name = T.name
+				TO.gender = T.gender	// Need to grab this too so PLURAL works properly in examine.
+				TO.opacity = FALSE
+				TO.plane = t_target
+		else if (T.below.bound_overlay)
+			QDEL_NULL(T.below.bound_overlay)
```
Comment (afterthought2) [inline code/controllers/subsystems/zcopy.dm:177]: Maybe cleaner to do
```
if(T.z_flags & ZM_MIMIC_OVERWRITE)
    // This openturf doesn't care about its icon, so we can just overwrite it.
    etc
else if(T.below.z_eventually_space)
    QDEL_NULL(T.below.bound_overlay)
else
    // Some openturfs have icons, so we can't overwrite their appearance.
    etc
```
though I could be misreading the logic branching here.

File: code/controllers/subsystems/zcopy.dm (hunk 1 above)
Comment (afterthought2) [reply code/controllers/subsystems/zcopy.dm:90]: Seems like they now go through `INCREMENT_WORLD_Z_SIZE`, so at least not hard to find. I'm not super clear on the ordering requirements for this; the connectivity is set in landmark/map_data/New, but if maps are being loaded via dmm_suite and queuing zmimic updates and then sleeping, there's no guarantee that will have been created before things start being queued here, unless you're already handling that. If you are, then can e.g. call this from both that macro and from `map_data/New`.

File: code/controllers/subsystems/zcopy.dm (hunk 2 above)
Comment (Lohikar) [reply code/controllers/subsystems/zcopy.dm:177]: That should work too; this is structured like this as the code it was extracted from had another branch above the bound_overlay check.

File: code/controllers/subsystems/zcopy.dm (hunk 1 above)
Comment (Lohikar) [reply code/controllers/subsystems/zcopy.dm:90]: This should only matter for things mutating the world size after Z-Mimic init, if away sites are loaded earlier than that they're probably fine.

Comment (afterthought2) [reply code/controllers/subsystems/zcopy.dm:90]: Well, away sites (including multiz ones) can be loaded via admin verb. There are a couple of other things that add blank z levels.

**Review by afterthought2 — COMMENTED** [review]
As discussed on discord, should refresh the `zstack` stuff in `New` of the map_data landmark. Otherwise looks good.

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**